application/comit_node/src/swap_protocols/rfc003/bob/spawner.rs

```diff
@@ -36,49 +36,38 @@ pub trait BobSpawner: Send + Sync + 'static {
     fn spawn<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset>(
         &self,
         id: SwapId,
-        swap_request: SwapRequest<AL, BL, AA, BA>,
-    ) -> Result<Box<ResponseFuture<Bob<AL, BL, AA, BA>>>, Error>
+        swap_request: rfc003::messages::Request<AL, BL, AA, BA>,
+    ) -> Result<Box<ResponseFuture<AL, BL>>, Error>
     where
         LedgerEventDependencies: CreateLedgerEvents<AL, AA> + CreateLedgerEvents<BL, BA>,
-        SwapRequest<AL, BL, AA, BA>: Into<Metadata>;
+        rfc003::messages::Request<AL, BL, AA, BA>: Into<Metadata>;
 }
 
 impl<T: MetadataStore<SwapId>, S: StateStore<SwapId>> BobSpawner for ProtocolDependencies<T, S> {
     #[allow(clippy::type_complexity)]
     fn spawn<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset>(
         &self,
         id: SwapId,
-        swap_request: SwapRequest<AL, BL, AA, BA>,
-    ) -> Result<Box<ResponseFuture<Bob<AL, BL, AA, BA>>>, Error>
+        swap_request: rfc003::messages::Request<AL, BL, AA, BA>,
+    ) -> Result<Box<ResponseFuture<AL, BL>>, Error>
     where
         LedgerEventDependencies: CreateLedgerEvents<AL, AA> + CreateLedgerEvents<BL, BA>,
-        SwapRequest<AL, BL, AA, BA>: Into<Metadata>,
+        rfc003::messages::Request<AL, BL, AA, BA>: Into<Metadata>,
     {
-        let save_state = self
-            .state_store
-            .new_save_state(id)
-            .map_err(Error::Storage)?;
+        let swap_seed = Arc::new(self.seed.swap_seed(id));
+        let bob = bob::State::new(swap_request.clone(), swap_seed);
+
+        let response_future = bob
+            .response_future()
+            .expect("bob::State is created in the SwapCommunication::Proposed variant");
```

Comment:
:thinking: The error message knows quite something about the internal state of `Bob` :)

Not sure if we can/should change that.